api/envoy/extensions/http/stateful_session/header/v3/header.proto

```diff
@@ -24,19 +24,48 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // <envoy_v3_api_msg_extensions.filters.http.stateful_session.v3.StatefulSession>`.
 //
 // For example, if the header name is set to ``session-header``, Envoy will prefer ``1.2.3.4:80``
-// as the upstream host when the request contains the following header:
+// as the upstream host when the request contains the following header (Note that plain text is used
+// here as example but actually header value will be base64 encoded):
 //
 // .. code-block:: none
 //
-//     session-header: "MS4yLjMuNDo4MA=="
+//     session-header: "1.2.3.4:80"
 //
 // When processing the upstream response, if ``1.2.3.4:80`` is indeed the final choice the extension
 // does nothing. If ``1.2.3.4:80`` is not the final choice, the new selected host will be set to
 // response headers (via the ``session-header`` response header).
 //
 // [#extension: envoy.http.stateful_session.header]
 message HeaderBasedSessionState {
-  // The name that will be used to obtain header value from downstream HTTP request or generate
-  // new header for downstream.
+  enum Mode {
+    // If an upstream host is selected and the its address is different from the request header value
```

Comment:
Can you clarify the cases for "If an upstream host is selected"?
My reading of this is that if the given host in the `name` header no longer exists, then an upstream host will be selected, and the header value in the response will be set.
Is the comment referring to just this case or to some other case?

Reply:
> Can you clarify the cases for "If an upstream host is selected"? My reading of this is that if the given host in the `name` header no longer exists, then an upstream host will be selected, and the header value in the response will be set. Is the comment referring to just this case or to some other case?

Sorry, the case actually is pretty simple. We will always select a upstream host, no matter based on the `name` header value or load balancing algorithm.
So, what I mean here is if the final upstream host is different with the `name` header value (or `name` header not exist), we will set the `name` header into response.